tests/LongRunning/SpeechRecoAuthTokenRefreshTests.ts

```diff
@@ -204,3 +204,175 @@ test("AuthToken refresh works correctly", (done: jest.DoneCallback) => {
             });
     });
 }, 1000 * 60 * 25); // 25 minutes.
+
+test("AuthToken refresh works correctly for Translation Recognizer", (done: jest.DoneCallback) => {
+    // tslint:disable-next-line:no-console
+    console.info("Name: AuthToken refresh works correctly for Translation Recognizer");
+
+    if (!Settings.ExecuteLongRunningTestsBool) {
+        // tslint:disable-next-line:no-console
+        console.info("Skipping test.");
+        done();
+        return;
+    }
+
+    const req = {
+        headers: {
+            "Content-Type": "application/json",
+            [HeaderNames.AuthKey]: Settings.SpeechSubscriptionKey,
+        },
+        url: "https://" + Settings.SpeechRegion + ".api.cognitive.microsoft.com/sts/v1.0/issueToken",
+    };
+
+    let authToken: string;
+
+    request.post(req, (error: any, response: request.Response, body: any) => {
+        authToken = body;
+    });
+
+    WaitForCondition(() => {
+        return !!authToken;
+    }, () => {
+        // Pump valid speech and then silence until at least one speech end cycle hits.
+        const fileBuffer: ArrayBuffer = WaveFileAudioInput.LoadArrayFromFile(Settings.WaveFile);
+
+        const s: sdk.SpeechTranslationConfig = sdk.SpeechTranslationConfig.fromAuthorizationToken(authToken, Settings.SpeechRegion);
+        s.speechRecognitionLanguage = "en-US";
+        s.addTargetLanguage("de-DE");
+        objsToClose.push(s);
+
+        let pumpSilence: boolean = false;
+        let bytesSent: number = 0;
+        let streamStopped: boolean = false;
+
+        // Pump the audio from the wave file specified with 1 second silence between iterations infinitely.
+        const p: sdk.PullAudioInputStream = sdk.AudioInputStream.createPullStream(
+            {
+                close: () => { return; },
+                read: (buffer: ArrayBuffer): number => {
+                    if (pumpSilence) {
+                        bytesSent += buffer.byteLength;
+                        if (bytesSent >= 32000) {
+                            bytesSent = 0;
+                            pumpSilence = false;
+                        }
+                        return buffer.byteLength;
+                    } else {
+                        const copyArray: Uint8Array = new Uint8Array(buffer);
+                        const start: number = bytesSent;
+                        const end: number = buffer.byteLength > (fileBuffer.byteLength - bytesSent) ? (fileBuffer.byteLength - 1) : (bytesSent + buffer.byteLength - 1);
+                        copyArray.set(new Uint8Array(fileBuffer.slice(start, end)));
```

Comment:
is it necessary to "new" an array, here? Won't slice hand that off?

Reply:
slice() gives an ArrayBuffer, which confusingly is not ArrayLike. (set() expects an ArrayLike parameter).